### PR TITLE
Refactored code to better support depndency injection.

### LINKED ISSUE #191: Investigate using Generic Host Builder for DI and config loading
.NET Core has generic [Host Builders](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.ihostbuilder?view=dotnet-plat-ext-3.1) which adds support for config loading and DI in the same way as ASP.NET Core (i.e using ConfigureServices).

### PR DESCRIPTION
**Pull request by boarnoah**
closes #191 
Ditto for internals of Service and Identity Providers. Refactored their calling of external APIs behind classes, make it easier to inject with mocks for unit testing.

### CODE REVIEW

File: Bmx.Service.Aws/AwsStsClient.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+
+namespace Bmx.Service.Aws {
+	internal class AwsStsClient : IAwsStsClient {
```
Comment (cfbao) [inline Bmx.Service.Aws/AwsStsClient.cs:7]: Is this just a wrapper around `AmazonSecurityTokenServiceClient`? Why not directly use [IAmazonSecurityTokenService](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SecurityToken/TISecurityTokenService.html)?

File: Bmx.Idp.Okta/OktaApi.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Bmx.Idp.Okta.Models;
+
+namespace Bmx.Idp.Okta {
+	internal class OktaApi : IOktaApi {
+		private readonly JsonSerializerOptions _serializeOptions =
+			new JsonSerializerOptions {PropertyNamingPolicy = JsonNamingPolicy.CamelCase,};
```
Comment (cfbao) [inline Bmx.Idp.Okta/OktaApi.cs:13]: nit: are you using [D2LTeamSettings.vssettings](https://github.com/Brightspace/vssettings)? This doesn't seem like it..

File: Bmx.Idp.Okta/OktaClient.cs [hunk 3]
```diff
@@ -13,32 +8,26 @@
...
 		private OktaApp[] _oktaApps;
 
-		public OktaClient() {
-			_cookieContainer = new CookieContainer();
-			_httpClient = new HttpClient( new HttpClientHandler {CookieContainer = _cookieContainer} );
-			_httpClient.Timeout = TimeSpan.FromSeconds( 30 );
-			_httpClient.DefaultRequestHeaders.Accept.Add( new MediaTypeWithQualityHeaderValue( "application/json" ) );
+		public OktaClient( IOktaApi oktaApi = null ) {
+			if( oktaApi == null ) {
+				_oktaApi = new OktaApi();
```
Comment (cfbao) [inline Bmx.Idp.Okta/OktaClient.cs:19]: Not sure how I feel about this pattern... Normally I'd just always inject the dependency.

**Comment by cfbao** [comment]
Overall looks good to me. I like having `IOktaApi` as a separate layer.
Have some comments on the specifics of DI handling. See also [this comment](https://github.com/Brightspace/bmx/issues/191#issuecomment-656931838) about using .NET Core DI.

File: Bmx.Idp.Okta/OktaApi.cs (hunk 2 above)
Comment (boarnoah) [reply Bmx.Idp.Okta/OktaApi.cs:13]: No, I got a modified version of the .editorconfig from the monorepo. I removed some things from there to better align it with the Roslyn defaults (what Microsoft recommends).

**Pull request merged**